### ISSUE TITLE
warning about bundleIdentifier is contradictory to prebuild workflow

### ISSUE THREAD

**Issue by ericvicenti** [expo/eas-cli#1292]
### Summary

Another warning fix, not urgent. When I run a build the following warning happens:

`Specified value for "ios.bundleIdentifier" in app.json is ignored because an ios directory was detected in the project.
EAS Build will use the value found in the native code.`

But this field is required when running `expo prebuild`, so I don't think I should be warned by eas CLI

Instead I think it would make sense to throw an error if the bundleIdentifier does not align between the native build and the app.json/expo.ios.bundleIdentifier

Thank you for smoothing out this experience!

### Managed or bare?

bare

### Environment

n/a

### Reproducible demo or steps to reproduce from a blank project

n/a

**Comment by wkozyra95** [comment]
It's very common that someone runs, `prebuild`/'run:ios`/`run:android` and does not realize that they "ejected", so it's important to keep this warning. 
This issue is only a problem when you have a bare project and use prebuild regularly, we could remove this contradiction, by modifying prebuilds to just prompt for bundleIdentifier value, but do not save it to app.json. This would just make that workflow harder to use, so it's probably not something you would want.

We discussed that problem internally in the past and decided on this change https://github.com/expo/eas-cli/commit/dd5afd354ab518a75e9214d64f3ccb99426d1288

**Comment by ericvicenti** [comment]
Ok so it's not possible for me to address this warning? That will make me less likely to notice or care next time I see a warning that comes from expo build

Why would the warning happen when the bundle IDs match?

I thought the bare workflow (expo prebuild) was a happy path now. But you seem to be implying I shouldn't use it regularly 🤷‍♂️

**Comment by wkozyra95** [comment]
> Ok so it's not possible for me to address this warning?

For a bare project no, but this is just a warning. This message is not intended to suggest that there is sth wrong with your project, our goal is to make sure that users are aware that values from app.json will not be used during build. If you think the wording of that message is not clear enough, I'm open to sugestions.

> Why would the warning happen when the bundle IDs match?

Even if they match this warning is still useful if someone is unintentionally ejected. I'm aware that it's not a good reason, but given how often this happens we think it's worth it.

> I thought the bare workflow (expo prebuild) was a happy path now. But you seem to be implying I shouldn't use it regularly

This is still a valid workflow, and we are supporting it, but in my opinion, a happy path for development with eas would be:
- Managed project, with gitignored android and ios directories, where prebuild would be used just for local testing. In this workflow, any native code changes would be ignored when building on EAS, but you can still have android/ios directories locally to run in Android studio/Xcode.
- Bare project, run prebuild only once at the start or at least it should be really rare e.g. when migrating between sdks. 

If changes in native code are small enough that you can easily reproduce them after running prebuild or running prebuild does not break them it should be easy to migrate that into config plugins and switch to managed workflow.

**Comment by awinograd** [comment]
👋 I'm popping in here from the discord in case this extra context is helpful. Essentially my vote would be to hide this warning when my ios/android folders are gitignored.

> awinograd — Today at 9:19 AM
> Anyone know of a way to run eas build without eas trying to derive bundleId from local android/ios folders? I run development builds locally but use eas for release builds. When I run eas build, it reads the bundle ID from my native folders even though they are gitignored
> Kim — Today at 9:21 AM
> If they are truly gitignored, they won't be uploaded, so that wouldn't be possible
> awinograd — Today at 9:21 AM
> the folder isn't uploaded but eas says it's trying to read some config info from the local folder. let me get the error msg
> Specified value for "ios.bundleIdentifier" in app.config.ts is ignored because an ios directory was detected in the project.
> EAS Build will use the value found in the native code.
> i guess i've never actually tried continuing the build once i see the warning. I just rm my folder and resubmit. perhaps it's just a "client" warning that isn't smart enough to know my folders are gitignored and when the build actually happens app.config.ts won't actually be ignored?
> Kim — Today at 9:38 AM
> Yes, that is actually the case
> It gives you that warning, because it sees that folder locally
> But it doesn't affect the build, as the folder isn't uploaded
> awinograd — Today at 9:38 AM
> awesome. that simplifies my workflow. thanks for the clarification

**Comment by wkozyra95** [comment]
> I'm popping in here from the discord in case this extra context is helpful. Essentially my vote would be to hide this warning when my ios/android folders are gitignored.

@awinograd When android and ios directories are gitignored this warning should not show up, if this is happening to you, please open a new issue.
